pkg/controller/base_controller.go

```diff
@@ -92,7 +99,15 @@ func (c *controller) Run(ctx context.Context) error {
 	wg.Add(c.numWorker)
 	for i := 0; i < c.numWorker; i++ {
 		go func() {
-			wait.Until(c.runWorker, c.syncPeriod, ctx.Done())
+			wait.Until(c.runWorker, time.Second, ctx.Done())
+			wg.Done()
+		}()
+	}
+
+	if c.sync != nil {
+		wg.Add(1)
+		go func() {
+			wait.Until(c.sync, c.syncPeriod, ctx.Done())
```

Comment:
I don't see any wg.wait.. are we missing it here?

Reply:
yes. I'll update it.